src/Mq3cPixelization.cc

```diff
@@ -0,0 +1,346 @@
+        double v1 = (v0 + faceScale) + DILATION;
+        u0 -= DILATION;
+        v0 -= DILATION;
+        std::tie(u0, v0) = atanApproxInverse(u0, v0);
+        std::tie(u1, v1) = atanApproxInverse(u1, v1);
+        verts[0] = faceToSphere(face, u0, v0, FACE_COMP, FACE_CONST);
+        verts[1] = faceToSphere(face, u1, v0, FACE_COMP, FACE_CONST);
+        verts[2] = faceToSphere(face, u1, v1, FACE_COMP, FACE_CONST);
+        verts[3] = faceToSphere(face, u0, v1, FACE_COMP, FACE_CONST);
+        // Even face numbers have right-handed uv coordinate systems,
+        // odd face numbers have left-handed ones. This has to be taken
+        // into account when generating pixel vertices, since convex
+        // polygon vertices must be specified in counter-clockwise order.
+        if ((face & 1) == 0) {
+            std::swap(verts[1], verts[3]);
+        }
+    }
+#else
+    void makeQuad(uint64_t i, int level, UnitVector3d * verts) {
+        int const face = static_cast<int>(i >> (2 * level)) - 10;
+        __m128d faceScale = _mm_set1_pd(FACE_SCALE[level]);
+        __m128d dilation = _mm_set1_pd(DILATION);
+        __m128d u0v0 = gridToFace(level, hilbertIndexInverseSimd(i, level));
+        __m128d u1v1 = _mm_add_pd(u0v0, faceScale);
+        u0v0 = atanApproxInverse(_mm_sub_pd(u0v0, dilation));
+        u1v1 = atanApproxInverse(_mm_add_pd(u1v1, dilation));
+        verts[0] = faceToSphere(face, u0v0, FACE_COMP, FACE_CONST);
+        verts[1] = faceToSphere(face, _mm_shuffle_pd(u1v1, u0v0, 2),
+                                FACE_COMP, FACE_CONST);
+        verts[2] = faceToSphere(face, u1v1, FACE_COMP, FACE_CONST);
+        verts[3] = faceToSphere(face, _mm_shuffle_pd(u0v0, u1v1, 2),
+                                FACE_COMP, FACE_CONST);
+        if ((face & 1) == 0) {
+            std::swap(verts[1], verts[3]);
+        }
+    }
+#endif
+
+
+// `Mq3cPixelFinder` locates modified-Q3C pixels that intersect a region.
+//
+// For now, we always begin with a loop over the root cube faces. For small
+// regions, this could be made significantly faster by computing the modified
+// Q3C index of the region bounding circle center, and looping over that pixel
+// and its neighbors.
+//
+// The subdivision level for the initial index computation would have to be
+// chosen such that the 8 or 9 pixel neighborhood of the center pixel is
+// guaranteed to contain the bounding circle. The minimum angular pixel width
+// could be precomputed per level. Alternatively, there is some constant W
+// such that the minimum angle between two points separated by at least one
+// pixel is greater than W * 2^-L at level L. Given the bounding circle
+// radius R, the subdivision level L of the initial neighborhood is the binary
+// exponent of W/R (and can be extracted by calling std::frexp).
+//
+// Finding W and implementing the above is left as a future optimization.
+template <typename RegionType, bool InteriorOnly>
```

Comment:
See `src/PixelFinder.h` - making it part of the type means some of the code in `visit` and `_insert` can be evaluated (or eliminated) at compiler time rather than at run-time.